packages/pyolite-kernel/src/worker.ts

```diff
@@ -1,420 +1,575 @@
+  EBADF: 8,
+  EBADFD: 127,
+  EBADMSG: 9,
+  EBADR: 114,
+  EBADRQC: 103,
+  EBADSLT: 102,
+  EBFONT: 101,
+  EBUSY: 10,
+  ECANCELED: 11,
+  ECHILD: 12,
+  ECHRNG: 106,
+  ECOMM: 124,
+  ECONNABORTED: 13,
+  ECONNREFUSED: 14,
+  ECONNRESET: 15,
+  EDEADLK: 16,
+  EDEADLOCK: 16,
+  EDESTADDRREQ: 17,
+  EDOM: 18,
+  EDOTDOT: 125,
+  EDQUOT: 19,
+  EEXIST: 20,
+  EFAULT: 21,
+  EFBIG: 22,
+  EHOSTDOWN: 142,
+  EHOSTUNREACH: 23,
+  EIDRM: 24,
+  EILSEQ: 25,
+  EINPROGRESS: 26,
+  EINTR: 27,
+  EINVAL: 28,
+  EIO: 29,
+  EISCONN: 30,
+  EISDIR: 31,
+  EL2HLT: 112,
+  EL2NSYNC: 156,
+  EL3HLT: 107,
+  EL3RST: 108,
+  ELIBACC: 129,
+  ELIBBAD: 130,
+  ELIBEXEC: 133,
+  ELIBMAX: 132,
+  ELIBSCN: 131,
+  ELNRNG: 109,
+  ELOOP: 32,
+  EMFILE: 33,
+  EMLINK: 34,
+  EMSGSIZE: 35,
+  EMULTIHOP: 36,
+  ENAMETOOLONG: 37,
+  ENETDOWN: 38,
+  ENETRESET: 39,
+  ENETUNREACH: 40,
+  ENFILE: 41,
+  ENOANO: 104,
+  ENOBUFS: 42,
+  ENOCSI: 111,
+  ENODATA: 116,
+  ENODEV: 43,
+  ENOENT: 44,
+  ENOEXEC: 45,
+  ENOLCK: 46,
+  ENOLINK: 47,
+  ENOMEDIUM: 148,
+  ENOMEM: 48,
+  ENOMSG: 49,
+  ENONET: 119,
+  ENOPKG: 120,
+  ENOPROTOOPT: 50,
+  ENOSPC: 51,
+  ENOSR: 118,
+  ENOSTR: 100,
+  ENOSYS: 52,
+  ENOTBLK: 105,
+  ENOTCONN: 53,
+  ENOTDIR: 54,
+  ENOTEMPTY: 55,
+  ENOTRECOVERABLE: 56,
+  ENOTSOCK: 57,
+  ENOTSUP: 138,
+  ENOTTY: 59,
+  ENOTUNIQ: 126,
+  ENXIO: 60,
+  EOPNOTSUPP: 138,
+  EOVERFLOW: 61,
+  EOWNERDEAD: 62,
+  EPERM: 63,
+  EPFNOSUPPORT: 139,
+  EPIPE: 64,
+  EPROTO: 65,
+  EPROTONOSUPPORT: 66,
+  EPROTOTYPE: 67,
+  ERANGE: 68,
+  EREMCHG: 128,
+  EREMOTE: 121,
+  EROFS: 69,
+  ESHUTDOWN: 140,
+  ESOCKTNOSUPPORT: 137,
+  ESPIPE: 70,
+  ESRCH: 71,
+  ESRMNT: 123,
+  ESTALE: 72,
+  ESTRPIPE: 135,
+  ETIME: 117,
+  ETIMEDOUT: 73,
+  ETOOMANYREFS: 141,
+  ETXTBSY: 74,
+  EUNATCH: 110,
+  EUSERS: 136,
+  EWOULDBLOCK: 6,
+  EXDEV: 75,
+  EXFULL: 115,
+};
 
-/**
- * Recursively convert a Map to a JavaScript object
- * @param The Map object to convert
- */
-function mapToObject(obj: any) {
-  const out: any = obj instanceof Array ? [] : {};
-  obj.forEach((value: any, key: string) => {
-    out[key] =
-      value instanceof Map || value instanceof Array ? mapToObject(value) : value;
-  });
-  return out;
-}
+export class PyoliteRemoteKernel {
+  constructor() {
+    this._initialized = new Promise((resolve, reject) => {
+      this._initializer = { resolve, reject };
+    });
+  }
 
-/**
- * Format the response from the Pyodide evaluation.
- *
- * @param res The result object from the Pyodide evaluation
- */
-function formatResult(res: any): any {
-  if (!pyodide.isPyProxy(res)) {
-    return res;
+  /**
+   * Accept the URLs from the host
+   **/
+  async initialize(options: IPyoliteWorkerKernel.IOptions): Promise<void> {
+    this._options = options;
+    await this.initRuntime(options);
+    await this.initPackageManager(options);
+    await this.initKernel(options);
+    await this.initGlobals(options);
+    await this.initFilesystem(options);
```

Comment:
Hm... I know i moved this down because i was breaking other stuff, but I wonder if we can move `initFilesystem` up to somwhere before `initKernel`: this would enable a _not_ not-invented-here way to do startup code for kernels, e.g. ipython profiles.